packages/database/src/migrations/20210603023401-AddFijiCOVIDVaccinePercentageOfPopulationVaccinatedMapOverlays-modifies-data.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+import { insertObject, generateId, codeToId } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const getOverlayId = (doseNum, level) =>
+  `FJ_COVID_TRACKING_Dose_${doseNum}_${level}_Percentage_Vaccinated`;
+
+const createMapOverlay = (doseNum, aggregationEntityType, level, overlayLevelName) => {
+  const numeratorDataElementCode = doseNum === 1 ? 'COVIDVac4' : 'COVIDVac8';
+  const denominatorAggregationEntityType =
+    aggregationEntityType === 'sub_district' ? undefined : aggregationEntityType;
+  const denominatorAggregationType =
+    aggregationEntityType === 'sub_district' ? undefined : 'SUM_PER_ORG_GROUP';
+
+  return {
+    id: getOverlayId(doseNum, level),
+    name: `% of Population Vaccinated COVID-19 Dose ${doseNum} (${overlayLevelName})`,
+    userGroup: 'COVID-19',
+    dataElementCode: 'value',
+    measureBuilderConfig: {
+      dataSourceType: 'custom',
+      measureBuilders: {
+        numerator: {
+          measureBuilder: 'sumAllPerOrgUnit',
+          measureBuilderConfig: {
+            aggregations: [
+              {
+                type: 'FINAL_EACH_DAY',
+              },
+            ],
+            dataElementCodes: [numeratorDataElementCode],
+            entityAggregation: {
+              dataSourceEntityType: 'facility',
+              aggregationEntityType,
+            },
+          },
+        },
+        denominator: {
+          measureBuilder: 'sumLatestPerOrgUnit',
+          measureBuilderConfig: {
+            dataElementCodes: ['population_FJ001'],
+            entityAggregation: {
+              dataSourceEntityType: 'sub_district',
+              aggregationType: denominatorAggregationType,
+              aggregationEntityType: denominatorAggregationEntityType,
+            },
+          },
+        },
+      },
+    },
+    measureBuilder: 'composePercentagePerOrgUnit',
```

Comment:
This migration is for: beyondessential/tupaia-backlog#2909
The overlay is shaded polygons at district/sub_district level. We want to get **latest for each day for a facility**, and aggregate (sum) to district/sub_district level. And calculate the percentage by dividing the **latest** population